src/Language/Fortran/Lexer/FixedForm.x

```diff
@@ -162,20 +216,24 @@ tokens :-
   <st,iif> "-"                                { addSpan TOpMinus  }
   <st,iif> "**"                               { addSpan TOpExp  }
   <st,iif> "*"                                { addSpan TStar  }
+    -- can be part of function type declaration
+  <keyword> "*" / { legacy77P }               { addSpan TStar  }
   <st,iif> "/"                                { addSpan TSlash  }
+  <st,iif> "&" / { legacy77P }                { addSpan TAmpersand  }
 
   -- Logical operators
   <st,iif> ".or."                             { addSpan TOpOr  }
   <st,iif> ".and."                            { addSpan TOpAnd  }
   <st,iif> ".not."                            { addSpan TOpNot  }
+  <st,iif> ".xor." / { legacy77P }            { addSpan TOpXOr  }
   <st,iif> ".eqv." / { fortran77P }           { addSpan TOpEquivalent  }
   <st,iif> ".neqv." / { fortran77P }          { addSpan TOpNotEquivalent  }
 
   -- Relational operators
   <st,iif> "<" / { extended77P }              { addSpan TOpLT  }
   <st,iif> "<=" / { extended77P }             { addSpan TOpLE  }
   <st,iif> "==" / { extended77P }             { addSpan TOpEQ  }
-  <st,iif> "!=" / { extended77P }             { addSpan TOpNE  }
+  <st,iif> "/=" / { extended77P }             { addSpan TOpNE  }
```

Comment:
As far as I can tell, `!=` is not a valid operator in f77. `gfortran` treats the `!` as starting an inline comment, and instead accepts `/=`.